transformers/avayaVoiceChannel/transformer.ts

```diff
@@ -0,0 +1,231 @@
+						break;
+					case 'prompt':
+						const menu = a.activityParams.menu;
+						sessionStorage['menu'] = menu;
+						output.text = (sessionStorage.cpaas_channel == 'sms') ? a.activityParams.text : ('<Say>' + a.activityParams.text + '</Say>');
+						sessionStorage.dtmf = true;
+						break;
+					case 'hangup':
+						sessionStorage.hangup = true;
+						break;
+					case 'play':
+						if (sessionStorage.cpaas_channel == 'call') {
+							output.text += '<Play>' + (a.activityParams.url ? a.activityParams.url  : "") + '</Play>';
+						} else {
+							output.text = a.activityParams.text;
+						}
+						break;
+					case 'record':
+						if (a.activityParams.shouldRecord) {
+							sessionStorage.record = '<Record method="POST" finishOnKey="#" action="'+a.activityParams.actionUrl+'"/>';
+						} else {
+							sessionStorage.record = null;
+						}
+						break;
+					case 'conference':
+						sessionStorage.conference = true;
+						let conferenceRoom = a.activityParams.conferenceRoom ? a.activityParams.conferenceRoom : "";
+						let confRoom = conferenceRoom.replace('+', '');
+						let conf = '<Dial><Conference startConferenceOnEnter="true" maxParticipants="2">' + confRoom + '</Conference></Dial>';
+						output.text = (output.text != null) ? (output.text + conf) : conf;
+						break;
+					default:
+						break;
+				}
+			});
+		} else if (output.text != null && (sessionStorage.cpaas_channel == 'call')) {
+			console.log('handleOutput: sessionStorage.cpaas_channel='.concat(sessionStorage.cpaas_channel));
+            output.text = '<Say>' + output.text + '</Say>';
+        }
+		return output;
+	},
+
+	/**
+	 * This transformer is executed when the Flow execution has finished.
+	 * For REST based transformers, the final output will be sent to
+	 * the user.
+	 *
+	 * @param processedOutput This is the final object that will be sent to the user.
+	 * It is therefore structured according to the Endpoint channel of the transformer.
+	 *
+	 * @param outputs This is an array of all of the outputs that were output by the Flow.
+	 * These will be merged to create the 'processedOutput' object.
+	 * 
+	 * @param userId The unique ID of the user.
+	 * @param sessionId The unique ID for this session. Can be used together with the userId
+	 * to retrieve the sessionStorage object.
+	 * @param endpoint The configuration object for the used Endpoint.
+	 * @param response The express response object that can be used to send a custom response back to the user.
+	 *
+	 * @returns An object that will be sent to the user, unchanged. It therefore has to have the
+	 * correct format according to the documentation of the specific Endpoint channel.
+	 */
+	handleExecutionFinished: async ({ processedOutput, outputs, userId, sessionId, endpoint, response }) => {
+		console.log("handleExecutionFinished called");
+		let url = COGNIGY_BASE_URL + endpoint.URLToken;
+		const sessionStorage = await getSessionStorage(userId, sessionId);
+		let r = 'default';
```

Comment:
Please change *r* to a better variable name. What does r stand for?

Reply:
made the changes per suggestion